source/board/microbitv2/microbitv2.c

```diff
@@ -52,3 +621,421 @@ const board_info_t g_board_info = {
+            if (size == 1) {
+                /* If size is 1 (only cmd id), this means it's a read */
+                uint32_t tempFileSize = __REV(gflashConfig.fileSize);
+                i2c_fillBuffer((uint8_t*) pI2cCommand, 0, 1);
+                i2c_fillBuffer((uint8_t*) &tempFileSize, 1, sizeof(gflashConfig.fileSize));
+            } else if (size == 5) {
+                /* If size is 5 (cmd id + 4B data), this means it's a write */
+                uint32_t tempFileSize = pI2cCommand->cmdData.data[0] << 24 |
+                                        pI2cCommand->cmdData.data[1] << 16 |
+                                        pI2cCommand->cmdData.data[2] << 8 |
+                                        pI2cCommand->cmdData.data[3] << 0;
+                
+                /* Validate file size */
+                if (tempFileSize <= (FLASH_INTERFACE_SIZE - FLASH_CONFIG_SIZE)) {
+                    gflashConfig.fileSize = tempFileSize;
+                    tempFileSize = __REV(gflashConfig.fileSize);
+                    i2c_fillBuffer((uint8_t*) pI2cCommand, 0, 1);
+                    i2c_fillBuffer((uint8_t*) &tempFileSize, 1, sizeof(gflashConfig.fileSize));
+                } else {
+                    pI2cCommand->cmdId = gFlashError_c;
+                    i2c_fillBuffer((uint8_t*) pI2cCommand, 0, 1);
+                }
+            } else {
+                pI2cCommand->cmdId = gFlashError_c;
+                i2c_fillBuffer((uint8_t*) pI2cCommand, 0, 1);
+            }
+        break;
+        case gFlashCfgFileVisible_c:
+            if (size == 1) {
+                /* If size is 1 (only cmd id), this means it's a read */
+                i2c_fillBuffer((uint8_t*) pI2cCommand, 0, 1);
+                i2c_fillBuffer((uint8_t*) &gflashConfig.fileVisible, 1, sizeof(gflashConfig.fileVisible));
+            } else if (size == 2) {
+                /* If size is 2 (cmd id + 1B data), this means it's a write */
+                gflashConfig.fileVisible = pI2cCommand->cmdData.data[0];
+                i2c_fillBuffer((uint8_t*) pI2cCommand, 0, 1);
+                i2c_fillBuffer((uint8_t*) &gflashConfig.fileVisible, 1, sizeof(gflashConfig.fileVisible));
+            } else {
+                pI2cCommand->cmdId = gFlashError_c;
+                i2c_fillBuffer((uint8_t*) pI2cCommand, 0, 1);
+            }
+        break;
+        case gFlashCfgWrite_c:
+            // Check first is config is already present in flash
+            // If differences are found, erase and write new config
+            if (0 != memcmp(&gflashConfig, (void *)FLASH_CONFIG_ADDRESS, sizeof(flashConfig_t))) {
+                status = erase_storage_sector(FLASH_CONFIG_ADDRESS);
+
+                if (status != 0) {
+                    pI2cCommand->cmdId = gFlashError_c;
+                }
+                else {
+                    status = program_storage_page(FLASH_CONFIG_ADDRESS, sizeof(flashConfig_t), (uint32_t *) &gflashConfig);
```

Comment:
I get an error on this line with gcc: ```
source/board/microbitv2/microbitv2.c:992:21: error: converting a packed 'flashConfig_t' {aka 'struct flashConfig_tag'} pointer (alignment 1) to a 'uint32_t' {aka 'long unsigned int'} pointer (alignment 4) may result in an unaligned pointer value [-Werror=address-of-packed-member]
  992 |                     status = program_storage_page(FLASH_CONFIG_ADDRESS, sizeof(flashConfig_t), (uint32_t *) &gflashConfig);
      |                     ^~~~~~
In file included from source/board/microbitv2/microbitv2.c:35:
source/board/microbitv2/i2c_commands.h:117:25: note: defined here
  117 | typedef __PACKED_STRUCT flashConfig_tag {
      |                         ^~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

Is there something to fix, or should I disable the warning?

Reply:
Was this addressed?

Reply:
Yes, in de84e4c.

Reply:
Perfect, thanks!